pkg/controller/plan/restore.go

```diff
@@ -360,6 +363,9 @@ func (r *RestoreManager) initFromAnnotation(synthesizedComponent *component.Synt
 	if doReadyRestoreAfterClusterRunning == "true" {
 		r.doReadyRestoreAfterClusterRunning = true
 	}
+	if env := backupSource[constant.EnvForRestore]; env != "" {
+		json.Unmarshal([]byte(env), &r.env)
```

Comment:
Return the error when unmarshal fails. It should rarely happen.

Reply:
If a error occurs, the `r.env` will be remained empty, should we still need to handle this err?

Reply:
If the env cannot be deserialized, we may not be able to restore the data as expected by the user (e.g., ignoring user-specified restore keys). The result of the restoration is unpredictable, so it is reasonable to fail the restoration directly.

Reply:
get